### PR TITLE
Add contributing instructions

### PR DESCRIPTION
**Pull request by ChanceNCounter**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1 +1,27 @@
...
+
+3. Your skill must be licensed under an OSI-approved open-source license, or released into the public domain, except by special permission from the OVOS dev team. Such permission is unlikely to be granted except in cases where your Skill integrates Mycroft or OVOS with software that is a) extremely valuable to users, b) extremely popular, c) privacy-respecting, and d) cannot be integrated any other way. (We're talking about things like a big-name streaming service, or integrating with a video game console. Big things with no open-source alternative. Submissions containing utility Skills under a proprietary license will be summarily closed as invalid.)
+
+Once you have completed these steps, you can create a pull request to this repository, adding your JSON file. Your Skill will be reviewed as soon as possible by one or more OVOS developers or authorized Skill Reviewers.
+
+Once your Skill has been reviewed, and the pull request is merged, your Skill should immediately be listed on the OVOS Skills Store, and become available for installation on any system running OSM!
+
+---
+
+On the JSON file:
```
Comment (JarbasAl) [inline readme.md:19]: NOTE: only url is required in the json file for `OSM`, in that case the branch is:
- implicit branch in url if applicable
- latest github release
- branch defined in skill.json file (if it exists in target github repo, default branch) 
- default branch of the repo

for acceptance in the appstore the .json must either include a `"branch"` field that corresponds to a github release, or have said release encoded in the url, e.g. `https://github.com/JarbasSkills/skill-ddg/tree/v0.1.0`

File: readme.md [hunk 2]
```diff
@@ -1 +1,27 @@
...
+
+* `branch` refers to the git tag mentioned above
+* `desktop_file` refers to an XDG-compliant desktop entry. Most Skills should leave this value `false`
+* `systemDeps` refers to operating system packages, which should be expressed as a list
+* `icon` should be a web-hosted .png
```
Comment (JarbasAl) [inline readme.md:24]: icon can be:
- url to a icon file
- relative path to github repo root
- auto detected in one of the [pre determined locations](https://github.com/OpenVoiceOS/ovos_skill_manager/blob/master/ovos_skills_manager/github/utils.py#L40)
- automatically extracted from readme

File: readme.md [hunk 3]
```diff
@@ -1 +1,27 @@
...
+* `folder` will literally become the name of the Skill's directory within your device's skills folder. The convention is `skillname.authorname`.
```
Comment (JarbasAl) [inline readme.md:25]: this parameter is meant for bigscreen support and usage is mostly dis encouraged , this is a workaround only to enable replacement of official mycroft skills

File: readme.md [hunk 4]
```diff
@@ -1 +1,27 @@
...
+* `categories` can list as many categories as you like, and the Skill will appear under each in the Skills Store. `category` refers to the Skill's *primary* category, which is the one that will appear next to its entry when clicked.
+* `tags` refers to other search terms you'd like to apply to this Skill
```
Comment (JarbasAl) [inline readme.md:27]: these greatly improve the search functionality of `osm` you are encouraged to add as many tags as you feel appropriate, this is one of the items reviewed for accuracy for accepting a skill

File: readme.md [hunk 5]
```diff
@@ -1 +1,27 @@
...
+Once you have completed these steps, you can create a pull request to this repository, adding your JSON file. Your Skill will be reviewed as soon as possible by one or more OVOS developers or authorized Skill Reviewers.
+
+Once your Skill has been reviewed, and the pull request is merged, your Skill should immediately be listed on the OVOS Skills Store, and become available for installation on any system running OSM!
+
+---
+
+On the JSON file:
+
+* `branch` refers to the git tag mentioned above
+* `desktop_file` refers to an XDG-compliant desktop entry. Most Skills should leave this value `false`
```
Comment (JarbasAl) [inline readme.md:22]: if set to True the desktop file is extracted from one of [supported locations](https://github.com/OpenVoiceOS/ovos_skill_manager/blob/master/ovos_skills_manager/github/utils.py#L45) or auto generated

distros like bigscreen usually launch a homescreen when the desktop icon is clicked, @AIIX  please expand this docs a bit

this is one of the items reviewed for accepting a skill, a homescreen must be present in the skill

File: readme.md (hunk 1 above)
Comment (JarbasAl) [reply readme.md:19]: this also means the url is the only required field for acceptance in skills store

File: readme.md (hunk 2 above)
Comment (ChanceNCounter) [reply readme.md:24]: need robust docs for this stuff in the appropriate places

File: readme.md (hunk 5 above)
Comment (ChanceNCounter) [reply readme.md:22]: can't put a TODO in here but TODO in addition to what i'm about to push, link to Plasma BigScreen skill dev docs

File: readme.md [hunk 6]
```diff
@@ -1 +1,27 @@
...
+Once you have completed these steps, you can create a pull request to this repository, adding your JSON file. Your Skill will be reviewed as soon as possible by one or more OVOS developers or authorized Skill Reviewers.
+
+Once your Skill has been reviewed, and the pull request is merged, your Skill should immediately be listed on the OVOS Skills Store, and become available for installation on any system running OSM!
+
+---
+
+On the JSON file:
+
+* `branch` refers to the git tag mentioned above, and can be omitted if the tag is in the URL.
+* `desktop_file` should be `true` if your Skill is associated with a FreeDesktop-compliant desktop entry. Most Skills should leave this `false`; Plasma BigScreen skills *must* include one
```
Comment (JarbasAl) [inline readme.md:22]: it is optional even for bigscreen! if it exists, the skill must have a homescreen (GUI that opens when you click the desktop icon), but a skill is not required to have homescreen, or .desktop file at all

Comment (ChanceNCounter) [reply readme.md:22]: ah, misunderstood

File: readme.md [hunk 7]
```diff
@@ -1 +1,27 @@
...
+
+Once your Skill has been reviewed, and the pull request is merged, your Skill should immediately be listed on the OVOS Skills Store, and become available for installation on any system running OSM!
+
+---
+
+On the JSON file:
+
+* `branch` refers to the git tag mentioned above, and can be omitted if the tag is in the URL.
+* `desktop_file` should be `true` if your Skill is associated with a FreeDesktop-compliant desktop entry. Most Skills should leave this `false`.
+* `systemDeps` refers to operating system packages, which should be expressed as a list
```
Comment (JarbasAl) [inline readme.md:23]: this is a boolean, it also comes from Pling, see https://plasma-bigscreen.org/development-guidelines/

requirements are defined in json file or manifest.yml/requirements.txt

**Comment by JarbasAl** [comment]
we can iterate over time, will just merge as is :+1: thanks for the initial write up

**Pull request merged**